cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/EquivalentBranchConversion.java

```diff
@@ -75,12 +80,74 @@ private void convertEquivalentBranchAtBoundary(int boundarySide) {
         double x = p.asDouble("x");
         double bch = 0;
         double gch = 0;
-        // If we have created buses and substations for boundary nodes,
-        // convert as a regular line
-        if (context.config().convertBoundary()) {
-            convertBranch(r, x, bch, gch);
+        // When the parameter convertBoundary is active:
+        // The boundary nodes are processed and included in the Network with a fictitious substation.
+        // The branches that lie on boundaries can be mapped to the Network directly in the ::convert method.
+        // So, this method should never be invoked when convertBoundary is active.
+        assert !context.config().convertBoundary();
```

Comment:
Could you throw here a more explicit exception in the (unlikely) case it happens?
```java
if (context.config().convertBoundary()) {
      throw new PowsyblException("Boundaries should be converted");
}
```

Reply:
The way the conversion works after we introducer handling boundaries using delayed nodes means that we will never reach this code when conversion of boundaries is active. This is why we preferred to put an assert instead of throwing an exception.

Reply:
But in this case, I'm not sure what the assert brings to the table? It can be simply deleted.

Reply:
As we have discussed, we changed the assert to an exception with a detailed explanation, just in case at some point the code is used with a different flow in the Conversion process.